modules/reindex/src/main/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSource.java

```diff
@@ -221,15 +222,23 @@ protected void doRun() throws Exception {
                 }
 
                 @Override
-                public void onFailure(Exception t) {
-                    if (t instanceof ResponseException) {
-                        ResponseException re = (ResponseException) t;
+                public void onFailure(Exception e) {
+                    if (e instanceof ResponseException) {
+                        ResponseException re = (ResponseException) e;
                         if (RestStatus.TOO_MANY_REQUESTS.getStatus() == re.getResponse().getStatusLine().getStatusCode()) {
-                            listener.onRetryableFailure(t);
+                            listener.onRetryableFailure(e);
                             return;
                         }
+                        /*
+                         * If possible wrap the ResponseException in an exception that'll preserve its status code so we can send it back to
+                         * the user.
+                         */
+                        RestStatus status = RestStatus.fromCode(re.getResponse().getStatusLine().getStatusCode());
```

Comment:
`RestStatus` is an enum so it only has the statuses that we've shoved in there. It is missing some of the weird, nonstandard ones like `418 I'm a teapot` and `420 Enhance Your Calm`. As weird as some of them are we can't just NPE on them, especially given that we could be going through a proxy. So, I don't wrap if we don't know about the error. You still get a message, but it gets squashed to a 500 error. That seems fair for rare, weird stuff. We can handle the issues if/when they come up.